File: src/Chapter05/SlickUpload-5.5.1/AdvancedSamples/AspNetMvcCS/UploadStatusModelBinder.cs
using System;
using System.Web;
using System.Web.Mvc;

using Krystalware.SlickUpload;
using Krystalware.SlickUpload.Controls;
using Krystalware.SlickUpload.Status;

namespace AspNetMvcCS
{
    public class UploadStatusModelBinder : IModelBinder
    {
        public static void Register(ModelBinderDictionary modelBinders)
        {
            if (!modelBinders.ContainsKey(typeof(UploadStatus)))
                modelBinders.Add(typeof(UploadStatus), new UploadStatusModelBinder());
        }

        public object BindModel(ControllerContext controllerContext, ModelBindingContext bindingContext)
        {
            UploadStatus status = HttpUploadModule.GetUploadStatus() ?? UploadConnector.GetUploadStatus();

            return status;
        }
    }
}

File: src/Chapter05/SlickUpload-5.5.1/AdvancedSamples/AspNetMvcVB/UploadStatusModelBinder.vb
Imports Krystalware.SlickUpload
Imports Krystalware.SlickUpload.Controls
Imports Krystalware.SlickUpload.Status

Public Class UploadStatusModelBinder
    Implements IModelBinder

    Public Shared Sub Register(ByVal modelBinders As ModelBinderDictionary)
        If Not modelBinders.ContainsKey(GetType(UploadStatus)) Then
            modelBinders.Add(GetType(UploadStatus), New UploadStatusModelBinder())
        End If
    End Sub

    Public Function BindModel(ByVal controllerContext As System.Web.Mvc.ControllerContext, ByVal bindingContext As System.Web.Mvc.ModelBindingContext) As Object Implements System.Web.Mvc.IModelBinder.BindModel
        Dim status As UploadStatus = HttpUploadModule.GetUploadStatus()

        If status Is Nothing Then
            status = UploadConnector.GetUploadStatus()
        End If

        Return status
    End Function
End Class
